internal/net/call/call_test.go

```diff
@@ -1343,7 +1343,7 @@ func TestCancelServe(t *testing.T) {
 	go func() {
 		opts := call.ServerOptions{Logger: logging.NewTestLogger(t)}
 		err := call.Serve(ctx, lis, handlers, opts)
-		if err != nil && ctx.Err() == nil {
+		if err != ctx.Err() {
```

Comment:
This function can never exit with `nil`, so I think what we want to check here is that the returned error is the `ctx.Err`, and not anything else.

Reply:
Yeah, this is much better.